### PR TITLE
Update default consul version 1.2.3->1.16.1

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -117,51 +117,7 @@ class { 'consul':
     end
 
     describe command('consul version') do
-      its(:stdout) { is_expected.to match %r{Consul v1.2.0} }
```
Comment (bastelfreak) [inline spec/acceptance/class_spec.rb:120]: the whole block was a duplicate, we don't need to test it multiple times.

File: manifests/init.pp [hunk 2]
```diff
@@ -137,7 +137,7 @@
   Enum['stopped', 'running']            $service_ensure              = 'running',
   Hash                                  $services                    = {},
   String[1]                             $user                        = $consul::params::user,
-  String[1]                             $version                     = '1.2.3',
+  String[1]                             $version                     = '1.16.1',
```
Comment (smortex) [inline manifests/init.pp:140]: I generally wonder if it is not better to _not_ set any default version, and mandate the user to explicitly put what they want.  This mean that we have to set an arbitrary version for testing, but we do not have to live with having to choose to default to an old-legacy-deprecated version after a few years or break backwards compatibility when a new version is available.

**Review by smortex — COMMENTED** [review]
See in-line comment, but if not appropriate with this module (I am not a user of consul) I am fine with this.

File: manifests/init.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/init.pp:140]: I considered that and I don't know if I like it or not. We need to do a major release anyway so I thought it's a good time to bump the version, also consul supports upgrades between versions.

**Review by smortex — APPROVED** [review]

**Pull request merged**